### PR TITLE
Add FpML parsing for CDS and CDX

### PR DESCRIPTION
**Pull request by jodastephen**
Ensure lots of sample FpML files parse

### CODE REVIEW

File: modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            .orElse(null));
+    XmlElement fixedAmountCalcEl = periodicPaymentEl.getChild("fixedAmountCalculation");
+    double fixedRate = document.parseDecimal(fixedAmountCalcEl.getChild("fixedRate"));
+    DayCount dayCount = fixedAmountCalcEl.findChild("dayCountFraction")
+        .map(el -> document.parseDayCountFraction(el))
+        .orElse(DayCounts.ACT_360);
+
+    // handle a single protectionTerms element
+    XmlElement protectionTermEl = cdsEl.getChild("protectionTerms");
+    CurrencyAmount notional = document.parseCurrencyAmount(protectionTermEl.getChild("calculationAmount"));
```
Comment (yukiiwashita) [inline modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java]: Is it helpful to check the notional values in the two fields are the same?

File: modules/product/src/main/java/com/opengamma/strata/product/credit/CdsIndex.java [hunk 2]
```diff
@@ -112,35 +114,45 @@
...
+   * When building, this will default to 'Beginning'.
    */
   @PropertyDefinition(validate = "notNull")
   private final ProtectionStartOfDay protectionStart;
   /**
    * The number of days between valuation date and step-in date.
    * <p>
    * The step-in date is also called protection effective date.
-   * It is usually 1 calendar day for standardized CDS index contracts.
+   * It is usually 1 calendar day for standardized CDS contracts.
```
Comment (yukiiwashita) [inline modules/product/src/main/java/com/opengamma/strata/product/credit/CdsIndex.java]: CDS -> CDS index

File: modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import com.opengamma.strata.product.credit.CdsTrade;
+
+/**
+ * FpML parser for CDS.
+ * <p>
+ * This parser handles the subset of FpML necessary to populate the trade model.
+ * <p>
+ * The following features are not available in the Strata trade model:
+ * <ul>
+ * <li>initial fixing date
```
Comment (yukiiwashita) [inline modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java]: These comments are relevant for CDSs?

File: modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java (hunk 1 above)
Comment (jodastephen) [inline modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java]: I think we can ignore the issue for now.

**Pull request merged**